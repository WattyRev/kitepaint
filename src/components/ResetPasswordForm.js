import React from "react";
import PropTypes from "prop-types";
import { A, P, Label, Input, Button } from "../theme";

const ResetPasswordForm = ({
  email,
  id,
  isDisabled,
  onCancel,
  onEmailChange,
  onSubmit,
  onUsernameChange,
  username
}) => (
  <form
    id={id}
    onSubmit={e => {
      e.preventDefault();
      onSubmit();
    }}
  >
    <P>
      *This is not hooked up yet.*Enter your username and email address to reset
      your password.
    </P>
    <Label htmlFor={`${id}-username`}>Username</Label>
    <Input
      id={`${id}-username`}
      value={username}
      disabled={isDisabled}
      onChange={e => onUsernameChange(e.target.value)}
    />
    <Label htmlFor={`${id}-email`}>Email Address</Label>
    <Input
      id={`${id}-email`}
      type="email"
      value={email}
      disabled={isDisabled}
      onChange={e => onEmailChange(e.target.value)}
    />
    <Button isPrimary isBlock type="submit">
      Reset Password
    </Button>
    <P>
      <A onClick={onCancel}>Cancel</A>
    </P>
  </form>
);

ResetPasswordForm.propTypes = {
  /**
   * The value for the email field.
   */
  email: PropTypes.string,
  /**
   * A unique identifier needed to differentiate instances of this componnet.
   */
  id: PropTypes.string.isRequired,
  /**
   * Indicates if the form is disabled.
   */
  isDisabled: PropTypes.bool,
  /**
   * A function called when the user clicks on the cancel button.
   */
  onCancel: PropTypes.func.isRequired,
  /**
   * A function called when the email changes. Is called with the new email as the first paramater.
   * @type {[type]}
   */
  onEmailChange: PropTypes.func.isRequired,
  /**
   * A function called when the form is submitted.
   */
  onSubmit: PropTypes.func.isRequired,
  /**
   * A function called when the username changes. Is called with the new username as the first parameter.
   */
  onUsernameChange: PropTypes.func.isRequired,
  /**
   * The value for the username field.
   */
  username: PropTypes.string
};

export default ResetPasswordForm;
