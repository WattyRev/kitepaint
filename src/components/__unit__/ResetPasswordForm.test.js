import React from "react";
import { shallow } from "enzyme";
import ResetPasswordForm from "../ResetPasswordForm";

describe("ResetPasswordForm", () => {
  let defaultProps;
  beforeEach(() => {
    defaultProps = {
      id: "abc",
      onCancel: jest.fn(),
      onEmailChange: jest.fn(),
      onSubmit: jest.fn(),
      onUsernameChange: jest.fn()
    };
  });
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(<ResetPasswordForm {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
