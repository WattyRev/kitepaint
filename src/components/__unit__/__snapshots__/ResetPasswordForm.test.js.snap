// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResetPasswordForm
    id="abc"
    onCancel={[MockFunction]}
    onEmailChange={[MockFunction]}
    onSubmit={[MockFunction]}
    onUsernameChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <P>
          *This is not hooked up yet.*Enter your username and email address to reset your password.
        </P>,
        <ForwardRef
          htmlFor="abc-username"
        >
          Username
        </ForwardRef>,
        <Input
          id="abc-username"
          onChange={[Function]}
        />,
        <ForwardRef
          htmlFor="abc-email"
        >
          Email Address
        </ForwardRef>,
        <Input
          id="abc-email"
          onChange={[Function]}
          type="email"
        />,
        <Button
          isBlock={true}
          isPrimary={true}
          type="submit"
        >
          Reset Password
        </Button>,
        <P>
          <ForwardRef
            onClick={[MockFunction]}
          >
            Cancel
          </ForwardRef>
        </P>,
      ],
      "id": "abc",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "*This is not hooked up yet.*Enter your username and email address to reset your password.",
        },
        "ref": null,
        "rendered": "*This is not hooked up yet.*Enter your username and email address to reset your password.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Username",
          "htmlFor": "abc-username",
        },
        "ref": null,
        "rendered": "Username",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
  margin: 8px 0 4px;
  display: block;
",
            ],
          },
          "displayName": "Styled(Label)",
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": undefined,
          "id": "abc-username",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Email Address",
          "htmlFor": "abc-email",
        },
        "ref": null,
        "rendered": "Email Address",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
  margin: 8px 0 4px;
  display: block;
",
            ],
          },
          "displayName": "Styled(Label)",
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": undefined,
          "id": "abc-email",
          "onChange": [Function],
          "type": "email",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Reset Password",
          "isBlock": true,
          "isPrimary": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Reset Password",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            onClick={[MockFunction]}
          >
            Cancel
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Cancel",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Cancel",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
              ],
            },
            "displayName": "Styled(A)",
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <P>
            *This is not hooked up yet.*Enter your username and email address to reset your password.
          </P>,
          <ForwardRef
            htmlFor="abc-username"
          >
            Username
          </ForwardRef>,
          <Input
            id="abc-username"
            onChange={[Function]}
          />,
          <ForwardRef
            htmlFor="abc-email"
          >
            Email Address
          </ForwardRef>,
          <Input
            id="abc-email"
            onChange={[Function]}
            type="email"
          />,
          <Button
            isBlock={true}
            isPrimary={true}
            type="submit"
          >
            Reset Password
          </Button>,
          <P>
            <ForwardRef
              onClick={[MockFunction]}
            >
              Cancel
            </ForwardRef>
          </P>,
        ],
        "id": "abc",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "*This is not hooked up yet.*Enter your username and email address to reset your password.",
          },
          "ref": null,
          "rendered": "*This is not hooked up yet.*Enter your username and email address to reset your password.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Username",
            "htmlFor": "abc-username",
          },
          "ref": null,
          "rendered": "Username",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
  margin: 8px 0 4px;
  display: block;
",
              ],
            },
            "displayName": "Styled(Label)",
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": undefined,
            "id": "abc-username",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Email Address",
            "htmlFor": "abc-email",
          },
          "ref": null,
          "rendered": "Email Address",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
  margin: 8px 0 4px;
  display: block;
",
              ],
            },
            "displayName": "Styled(Label)",
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": undefined,
            "id": "abc-email",
            "onChange": [Function],
            "type": "email",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Reset Password",
            "isBlock": true,
            "isPrimary": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Reset Password",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              onClick={[MockFunction]}
            >
              Cancel
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Cancel",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "rules": Array [
                  "
  color: ",
                  [Function],
                  ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
                ],
              },
              "displayName": "Styled(A)",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
